util/movement-types/src/lib.rs

```diff
@@ -118,27 +159,55 @@ impl Block {
 		Self { metadata, parent, transactions, id }
 	}
 
-	pub fn id(&self) -> Id {
-		self.id.clone()
+	pub fn into_parts(self) -> (BlockMetadata, Vec<u8>, BTreeSet<Transaction>, Id) {
+		(self.metadata, self.parent, self.transactions, self.id)
+	}
+
+	pub fn id(&self) -> &Id {
+		&self.id
+	}
+
+	pub fn parent(&self) -> &Vec<u8> {
+		&self.parent
+	}
+
+	pub fn transactions(&self) -> &BTreeSet<Transaction> {
```

Comment:
This exposes the internal collection as part of the API.
Would it be better to return an iterator?

If you move the `Block` definitions in `mod block` so that the iterator can be named `block::Transactions`:

```rust
use crate::Transaction;
use std::collections::btree_set;

pub type Transactions<'a> = btree_set::Iter<'a, Transaction>;

impl Block {
    pub fn transactions(&self) -> Transactions {
        self.transactions.iter()
    }
}
```

Reply:
To get it even more encapsulated, have the iterator as a newtype wrapping the inner `btree_set::Iter` and implement `Iterator`, `DoubleEndedIterator`, `ExactSizeIterator`, and all the rest of the iterator marker traits on the newtype, but it's too much boilerplate for now, so maybe leave that for 1.0.

Reply:
So, just the `pub type Transactions<'a> = btree_set::Iter<'a, Transaction>` is your recommendation?

Reply:
Yes, that would do for now. I would split up into `block`, `transaction`, etc. modules already though, so we can have path-distinguished naming for these subsidiary types, rather than long C-programmers-disease affected names at the crate level.

Reply:
Sounds good!